src/gui/addnewtorrentdialog.cpp

```diff
@@ -830,3 +828,11 @@ void AddNewTorrentDialog::doNotDeleteTorrentClicked(bool checked)
 {
     m_torrentGuard->setAutoRemove(!checked);
 }
+
+void AddNewTorrentDialog::openDialog()
+{
+#ifdef Q_OS_MAC
+    setModal(true);
```

Comment:
This can be moved to constructor, then the openDialog() will become trivial and replaced by calling open() directly.